packages/tangle/ledgerstate.go

```diff
@@ -54,7 +54,6 @@ func (l *LedgerState) InheritBranch(referencedBranchIDs ledgerstate.BranchIDs) (
 	if err != nil {
 		if errors.Is(err, ledgerstate.ErrInvalidStateTransition) {
 			inheritedBranch = ledgerstate.InvalidBranchID
-			err = nil
```

Comment:
We need to properly revisit the topic of invalid and lazy booked messages since there are too many changes with this PR. That's why we created this issue: https://github.com/iotaledger/goshimmer/issues/1587

Reply:
I don't see how any of the changes affects the behavior of lazy booked messages and I also don't understand why "too many changes" justifies even more unrelated changes that we now need to possibly revert again in a future PR.

Wouldn't it be better to create a separate issue for revisiting the topic of lazy booking all together instead of introducing agreement failures in the protocol?